pxr/usd/ar/resolver.cpp

```diff
@@ -1297,10 +1302,16 @@ class _DispatchingResolver final
         if (uriResolver) {
             return *uriResolver;
         }
+
+        if (!_resolver)
+        {
+            TF_FATAL_CODING_ERROR("Resolver is uninitialized.");
```

Comment:
We are very cautious with use of fatal errors in OpenUSD, because as a generally embedded subsystem, it means we're taking down somebody else's DCC.  I believe the *only* places where TF_FATAL_CODING_ERROR is issued in OpenUSD, currently, is in the Metal integration, and we might want to reconsider those, too?

Also, here and above, is it clear that it is a coding error?  I imagine a resolver might fail to initialize because, e.g. a database went down somewhere.  So maybe these should both be TF_WARN. (and in the db case, possibly the resolver itself might have already reported a TF_RUNTIME_ERROR if it could not contact the db).

I realize leaving this as a warning does still likely leave unguarded access to _resolver in some other sites, here, but perhaps we should harden those, knowing that an error/warning has already been issued?

Reply:
That's fair. I figured an explicit abort was better than an unintended segfault, but I changed it to a TF_WARN instead as requested.

I agree that it's unclear on why the resolver is uninitialized, though I was unsure what the best way to harden this access is. 
We tend to hit this one when we run any of the CLI targets from the IDE and it can't find the plugInfo.json etc to initialize (since it's not installed yet).